### PR TITLE
mysql 8 upgrade

### PR DESCRIPTION
**Pull request by yonghaoy**
Description:

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/db-cdr/build.gradle [hunk 1]
```diff
@@ -17,7 +17,7 @@ dependencies {
     liquibaseRuntime 'org.liquibase:liquibase-core:3.10.0'
     liquibaseRuntime 'org.liquibase:liquibase-groovy-dsl:3.0.0'
     liquibaseRuntime 'info.picocli:picocli:4.6.1'
-    liquibaseRuntime 'mysql:mysql-connector-java:5.1.37'
```
Comment (yonghaoy) [inline api/db-cdr/build.gradle:20]: Find solution here: 
https://forum.liquibase.org/t/liquibaseexception-java-lang-classcastexception-class-java-time-localdatetime-cannot-be-cast-to-class-java-lang-string/5059/5

File: api/build.gradle [hunk 2]
```diff
@@ -639,10 +639,10 @@ def db_host = System.getenv("DB_HOST")
 def db_port = System.getenv("DB_PORT")
 
 def commandLineSpringProperties = [
-        'spring.datasource.driver-class-name': 'com.mysql.jdbc.Driver',
+        'spring.datasource.driver-class-name': 'com.mysql.cj.jdbc.Driver',
         // Our database isn't configured to allow outside connections (which is good),
         // so our connections are either local or through the encrypted proxy.
-        'spring.datasource.url'              : "jdbc:mysql://${db_host}:${db_port}/workbench?useSSL=false",
+        'spring.datasource.url'              : "jdbc:mysql://${db_host}:${db_port}/workbench?useSSL=false&allowPublicKeyRetrieval=true",
```
Comment (yonghaoy) [inline api/build.gradle:645]: other solutions: https://community.atlassian.com/t5/Confluence-questions/MySQL-Public-Key-Retrieval-is-not-allowed/qaq-p/778956

File: api/build.gradle [hunk 3]
```diff
@@ -403,7 +403,7 @@ dependencies {
     compile "io.swagger:swagger-annotations:1.5.16"
     compile "javax.inject:javax.inject:1"
     compile "org.liquibase:liquibase-core:3.10.3"
-    compile "mysql:mysql-connector-java:8.0.28"
+    compile "mysql:mysql-connector-java:8.0.22"
```
Comment (yonghaoy) [inline api/build.gradle:406]: if we want 8.0.28 back, we may need to upgrade liquibase to 4.3.x

Comment (jmthibault79) [reply api/build.gradle:406]: FYI https://security.snyk.io/vuln/SNYK-JAVA-MYSQL-1766958

File: api/db-cdr/generate-cdr/init-new-cdr-db.sh [hunk 4]
```diff
@@ -42,14 +42,7 @@ trap finish EXIT
...
 function run_mysql() {
-  if [ -f /.dockerenv ]; then
-    mysql $@
-  else
-    echo "Outside docker: invoking mysql via docker for portability"
-    docker run -i --rm --network host --entrypoint '' \
-      mysql:5.7.27 \
-      mysql $@
-  fi
+  mysql $@
```
Comment (jmthibault79) [inline api/db-cdr/generate-cdr/init-new-cdr-db.sh:45]: will running these commands require local non-docker mysql installs?

File: api/db/run-migrations.sh [hunk 5]
```diff
@@ -22,17 +23,11 @@ trap finish EXIT
...
-    docker run --rm --network host --entrypoint '' -i \
-      mysql:5.7.27 \
-      mysql $@
-  fi
+  mysql $@
 }
 
 echo "Creating database if it does not exist..."
+
 cat "${CREATE_DB_FILE}" | run_mysql -h "${DB_HOST}" --port "${DB_PORT}" -u root -p"${MYSQL_ROOT_PASSWORD}"
```
Comment (jmthibault79) [inline api/db/run-migrations.sh:31]: should this be `HEX_DB_PASSWORD` ?

**Review by jmthibault79 — COMMENTED** [review]
Probably also need to modify these references to 5.7.27:
api/libproject/devstart.rb
api/libproject/mysql_docker.rb

File: api/db/run-migrations.sh (hunk 5 above)
Comment (yonghaoy) [reply api/db/run-migrations.sh:31]: HEX_DB_PASSWORD was used in mariadb, for mysql, HEX_DB_PASSWORD is not needed

**Comment by yonghaoy** [comment]
I don't think Codacy gives me a good advide

File: api/docker-compose.yaml [hunk 6]
```diff
@@ -2,13 +2,16 @@ version: "3.4"
...
-    image: mysql:5.7.27
+    platform: linux/amd64
+    image: mysql:8.0.27
     env_file:
       - db/vars.env
     volumes:
       - db:/var/lib/mysql
     ports:
       - 127.0.0.1:3306:3306
+    cap_add:
```
Comment (calbach) [inline api/docker-compose.yaml:13]: This change was needed on my machine, otherwise the DB service failed to start with "mbind: Operation not permitted"

**Pull request closed without merging**